FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -231,6 +231,17 @@ public function getFindologicArticle($shopwareArticle)
                     $shopwareArticle->getMainDetail()->getNumber()
                 )
             );
+        } catch (EntityNotFoundException $e) {
```

Comment:
Why aren't we just catching `\Exception` here instead? We do this for Shopware 6, but with the difference that we explicitly add `... If you see this message in your logs, please report this as a bug ...`:

https://github.com/findologic/plugin-shopware-6/blob/0dd54ada1ff1d46ab3febde8e52628b2fc0f9957/src/Export/XmlProduct.php#L258

In Shopware 6 we use `\Throwable`, but we can not do this for Shopware 5, as Shopware 5.2 and I also think 5.3 still support PHP 5.6.

Reply:
@TheKeymaster The suggestion to just catch `EntityNotFoundException` came from me. I thought it's maybe not clear from the customer-perspective when complete products are missing because we skip it always. But if we already use the same logic for SW6, I'll guess it's fine for SW5 as well.

Reply:
We can also separate them and add an additional catch for the `EntityNotFoundException`.

```php
// ...
} catch (EntityNotFoundException $e) {
    // ... log that the entity was not found somehow.
} catch (\Exception $e) {
   // ... log that something went wrong.
}
```